docs/syntax/syntax.md

```diff
@@ -205,95 +205,429 @@ Is below, but it won't be parsed into the document.
 
 (syntax/referencing)=
 
-## Markdown Links and Referencing
+## Links and Referencing
 
-Markdown links are of the form: `[text](link)`.
+CommonMark links come in three forms ([see the spec](https://spec.commonmark.org/0.30/#links)):
 
-If you set the configuration `myst_all_links_external = True` (`False` by default),
-then all links will be treated simply as "external" links.
-For example, in HTML outputs, `[text](link)` will be rendered as `<a href="link">text</a>`.
+*Autolinks* are [URIs][uri] surrounded by `<` and `>`:
 
-Otherwise, links will only be treated as "external" links if they are prefixed with a scheme,
-configured with `myst_url_schemes` (by default, `http`, `https`, `ftp`, or `mailto`).
-For example, `[example.com](https://example.com)` becomes [example.com](https://example.com).
+```md
+<scheme:path?query#fragment>
+```
 
-:::{note}
-The `text` will be parsed as nested Markdown, for example `[here's some *emphasised text*](https://example.com)` will be parsed as [here's some *emphasised text*](https://example.com).
+*Inline links* allow for optional explicit text and titles (in HTML titles are rendered as tooltips):
+
+```md
+[Explicit *Markdown* text](destination "optional explicit title")
+```
+
+or, if the destination contains spaces,
+
+```md
+[text](<a destination>)
+```
+
+*Reference links* define the destination separately in the document, and can be used multiple times:
+
+```md
+[Explicit *Markdown* text][label]
+
+[label]: destination "optional explicit title"
+```
+
+MyST, supports the following destination types:
+
+|     Link Type      |            Auto            |          Inline           | Single Page[^sp] |
+| :----------------- | :------------------------- | :------------------------ | :--------------: |
+| External URL       | `<https://example.com>`    | `[](https://example.com)` |        ✅         |
+| Local file path    | `<path:file.txt>`          | `[](file.txt)`            |        ❌         |
+| Project document   | `<project:file.md>`        | `[](file.md)`             |        ❌         |
```

Comment:
I also feel that this isn't yet a great name, we could also make this longer as it is unlikely to be widely used? `localproject:`? `local:`?

Of the above list I am most partial to `target:` (doc, ref, id and key, seem either too specific or too general to me).

Reply:
I like `target`. `<target#sometarget>` seems reasonable...maybe specifying the project name could be a kwarg? E.g., `<target?project=someproject#sometarget>`?

Reply:
> `<target#sometarget>` seems reasonable

Just to note, this is not valid Markdown (you'll see if you try it <target#sometarget>); auto-links are only recognised if they have a schema: <schema:target#sometarget>

Reply:
> Can we define a word that is more semantically tied to the thing being referenced? 

From https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax
> Each URI begins with a scheme name that refers to a specification for assigning identifiers within that scheme. As such, the URI syntax is a federated and extensible naming system wherein each scheme's specification may further restrict the syntax and semantics of identifiers using that scheme. 

I would say `project` should go to `myst`, because it is the "MyST specification" we are using to assign the identifiers